joblib/_parallel_backends.py

```diff
@@ -0,0 +1,220 @@
+"""
+Backends for embarrassingly parallel code.
+"""
+
+import gc
+import os
+import warnings
+import threading
+
+from ._multiprocessing_helpers import mp
+if mp is not None:
+    from .pool import MemmapingPool
+    from multiprocessing.pool import ThreadPool
+
+# Environment variables to protect against bad situations when nesting
+JOBLIB_SPAWNED_PROCESS = "__JOBLIB_SPAWNED_PARALLEL__"
+
+# In seconds, should be big enough to hide multiprocessing dispatching
+# overhead.
+# This settings was found by running benchmarks/bench_auto_batching.py
+# with various parameters on various platforms.
+MIN_IDEAL_BATCH_DURATION = .2
+
+# Should not be too high to avoid stragglers: long jobs running alone
+# on a single worker while other workers have no work to process any more.
+MAX_IDEAL_BATCH_DURATION = 2
+
+
+class ImmediateResult(object):
+    def __init__(self, batch):
+        # Don't delay the application, to avoid keeping the input
+        # arguments in memory
+        self.results = batch()
+
+    def get(self):
+        return self.results
+
+
+###############################################################################
+class SequentialBackend(object):
+
+    def __init__(self, parallel):
+        self.parallel = parallel
+
+    def effective_n_jobs(self, n_jobs):

```

Comment:
It would be good to have on liner docstrings on all these methods: it would help other developers understand what the philosophy/purpose is behind them.